packages/skeleton/src/lib/components/Autocomplete/types.ts

```diff
@@ -1,12 +1,12 @@
 // Autocomplete Types
 
-export interface AutocompleteOption<Value = unknown> {
+export interface AutocompleteOption<Value = unknown, Meta = unknown> {
 	/** Provide a unique display label per option. Supports HTML. */
 	label: string;
 	/** Provide a unique option value. */
 	value: Value;
 	/** Provide a comma separated list of keywords. */
-	keywords?: any;
+	keywords?: string;
 	/** Pass arbitrary data per option. */
-	meta?: any;
+	meta?: Meta;
```

Comment:
A DX improvement could be made that if meta value is provided then it can't be undefined like so
```ts
export type AutocompleteOption<Value = unknown, Meta = unknown> = {
	/** Provide a unique display label per option. Supports HTML. */
	label: string;
	/** Provide a unique option value. */
	value: Value;
	/** Provide a comma separated list of keywords. */
	keywords?: string;
} & ([undefined] extends [Meta]
	? {
			/** Pass arbitrary data per option. */
			meta?: Meta;
	  }
	: {
			/** Pass arbitrary data per option. */
			meta: Meta;
	  });
```

Not sure it that's necessary though and even if it is - if there's a nicer way to do that. Don't think there's a way to do it with interfaces.

Reply:
I don't think that'll be necessary since it'll likely be a breaking change for those that vary the type of their `meta` field. 

For instance, consider the following:
```ts
// note that we're not specifying a Meta type, so it defaults to `unknown`
type FlavorOption = AutocompleteOption<string>;
const flavorOptions: FlavorOption[] = [
	{ label: 'Vanilla', value: 'vanilla', keywords: 'plain, basic', meta: { healthy: false } },
	{ label: 'Chocolate', value: 'chocolate', keywords: 'dark, white', meta: { healthy: false } },
	{ label: 'Peach', value: 'peach', keywords: 'fruit' }
];
```
The last value in the array (Peach) will have a type error, despite `meta` being optional. It can be fixed by specifying the type for Meta in `AutocompleteOption`, but the default behavior will break.

Ideally, we'd want the default generic type of `Meta` to be `undefined`, but that's also going to result in breaking the types for people that have already implemented the `AutocompleteOption` type.

So for now, let's just keep it as is and I'll note that it should be changed to `undefined` in the next major version of Skeleton.